### ISSUE TITLE
[MU4 Issue] Crash when dragging hairpin under specific circumstances involving mmrest

### ISSUE THREAD

**Issue by cbjeukendrup** [musescore/MuseScore#13677]
**To Reproduce**
1. Create score (1 instrument is enough)
2. enter a tempo marking on measure 3 and 5, so that mmrests will be broken
    (alternatively: enter a ritenuto spanning measure 3 and 4)
3. enter a hairpin spanning measure 3 and 4
4. press <kbd>m</kbd> to enable multi-measure rests
5. select the hairpin
6. select the start grip of the hairpin
7. press Shift+left arrow (might need to do that a few times)
8. try to drag the hairpin
-> crash

**Screenshots**

https://user-images.githubusercontent.com/48658420/194776138-c92ac40d-2fd0-4e44-b7dc-8fdf637f7aa8.mov

**Cause**
`l->startSegment()->system()` is nullptr in https://github.com/musescore/MuseScore/blob/eb5174099da996369df43eddee7f51c0961fa1e2/src/engraving/libmscore/line.cpp#L668
Additionally, `l->startSegment()->explicitParent()` is a `Measure`, and the parent of that measure is a `DummyElement`.

**Stack trace**
```
#0 0x102103211 in std::__1::vector<mu::engraving::SysStaff*, std::__1::allocator<mu::engraving::SysStaff*> >::operator[](unsigned long) const vector:1581
#1 0x102111b6d in mu::engraving::System::staffCanvasYpage(unsigned long) const system.cpp:1457
#2 0x10171d07c in mu::engraving::LineSegment::rebaseAnchors(mu::engraving::EditData&, mu::engraving::Grip) line.cpp:673
#3 0x10171fbdf in mu::engraving::LineSegment::drag(mu::engraving::EditData&) line.cpp:811
#4 0x1011fa6fc in mu::engraving::SingleElementGroup::drag(mu::engraving::EditData&) elementgroup.cpp:34
#5 0x104910988 in mu::notation::NotationInteraction::drag(mu::PointX<double> const&, mu::PointX<double> const&, mu::notation::DragMode) notationinteraction.cpp:990
#6 0x1047309fc in mu::notation::NotationViewInputController::mouseMoveEvent(QMouseEvent*) notationviewinputcontroller.cpp:710
...
```

**Comment by Tantacrul** [comment]
Will this also be fixed by https://github.com/musescore/MuseScore/pull/13676 ?

**Comment by cbjeukendrup** [comment]
No, unfortunately not. It looks like this one is going to involve a bit more research. 

@mike-spa @asattely If you have any suggestions or would have time to take a look at this too, that would be absolutely welcome!

**Comment by cbjeukendrup** [comment]
The exact same crash also happens in MS3. 

I think I might know where it's going wrong, but not sure yet...

**Comment by cbjeukendrup** [comment]
Finally pushed a fix to #13676!
